### PR TITLE
[#2062][FOLLOWUP] improvement(dashboard): Display detailed info for application in dashboard ui

### PR DESCRIPTION
**Pull request by lwllvyb**
### What changes were proposed in this pull request?
Display detailed info for application in dashboard ui

### Why are the changes needed?
(Please clarify why the changes are needed. For instance,

If you propose a new API, clarify the use case for a new API.
If you fix a bug, describe the bug.)
Fix: https://github.com/apache/incubator-uniffle/issues/2062

### Does this PR introduce any user-facing change?
(Please list the user-facing changes introduced by your change, including

Change in user-facing APIs.
Addition or removal of property keys.)
Yes.

### How was this patch tested?
(Please test your changes, and provide instructions on how to test it:

If you add a feature or fix a bug, add a test to cover your changes.
If you fix a flaky test, repeat it for many times to prove it works.)

image

### CODE REVIEW

**Comment by maobaolong** [comment]
@zuston the spill api would be invoked while spark executor need evict memory

**Comment by zuston** [comment]
LGTM. Please resolve the conflicts and then I will merge this. Thanks @lwllvyb

File: storage/src/main/java/org/apache/uniffle/storage/common/AbstractStorage.java [hunk 1]
```diff
@@ -40,17 +41,22 @@ public abstract class AbstractStorage implements Storage {
   abstract ShuffleWriteHandler newWriteHandler(CreateShuffleWriteHandlerRequest request);
 
   @Override
-  public ShuffleWriteHandler getOrCreateWriteHandler(CreateShuffleWriteHandlerRequest request) {
+  public Pair<ShuffleWriteHandler, Boolean> getOrCreateWriteHandler(
```
Comment (zuston) [inline storage/src/main/java/org/apache/uniffle/storage/common/AbstractStorage.java:44]: I don't hope to change the original abstract definition, this pair returning structure looks tricky.

From my sight, why not making the is_created or is_cached as the internal var into the concreted ShuffleWriteHandler, and then expose this as a interface method?

Comment (lwllvyb) [reply storage/src/main/java/org/apache/uniffle/storage/common/AbstractStorage.java:44]: I will recheck the code base on your suggestion.

Comment (lwllvyb) [reply storage/src/main/java/org/apache/uniffle/storage/common/AbstractStorage.java:44]: I rechecked the code, but i didn't find a better way to fix it. Can you give me some code level suggestions?
@zuston

Comment (zuston) [reply storage/src/main/java/org/apache/uniffle/storage/common/AbstractStorage.java:44]:
```
class ShuffleWriteHandlerWrapper {
  private ShuffleWriteHandler handler;
  private boolean isNewlyCreated;
}


@Override
  public ShuffleWriteHandler getOrCreateWriteHandler(CreateShuffleWriteHandlerRequest request) {
    writerHandlers.computeIfAbsent(request.getAppId(), key -> JavaUtils.newConcurrentMap());
    requests.computeIfAbsent(request.getAppId(), key -> JavaUtils.newConcurrentMap());
    Map<String, ShuffleWriteHandler> map = writerHandlers.get(request.getAppId());
    String partitionKey =
        RssUtils.generatePartitionKey(
            request.getAppId(), request.getShuffleId(), request.getStartPartition());
    map.computeIfAbsent(partitionKey, key -> newWriteHandler(request));
    Map<String, CreateShuffleWriteHandlerRequest> requestMap = requests.get(request.getAppId());
    requestMap.putIfAbsent(partitionKey, request);
    ShuffleWriteHandler handler = map.get(partitionKey);
    return new ShuffleWriteHandlerWrapper(handler, true/false);
  }
```

like this

**Comment by lwllvyb** [comment]
image
There is a test case that failed, but it is not related to the PR.

File: storage/src/main/java/org/apache/uniffle/storage/common/AbstractStorage.java (hunk 1 above)
Comment (lwllvyb) [reply storage/src/main/java/org/apache/uniffle/storage/common/AbstractStorage.java]: Fixed

**Comment by lwllvyb** [comment]
Ping @zuston

**Comment by lwllvyb** [comment]
Ping @jerqi  @maobaolong @zuston 
Could you please take some time to review this PR? If it doesn't need to be merged, please let me know. This PR has been verified internally.

**Pull request closed without merging**